service/mantle/ups/UpsServices.xml

```diff
@@ -47,4 +47,70 @@
         </actions>
     </service>
 
+    <service verb="get" noun="AutoPackageInfoList">
+        <implements service="mantle.shipment.CarrierServices.get#AutoPackageInfo"/>
+        <actions>
+            <iterate list="itemInfoList" entry="itemInfoMap">
+                <entity-find entity-name="mantle.shipment.ShipmentItemSource" list="shipmentSourceList">
+                    <econdition field-name="orderId" from="itemInfoMap.orderId"/>
+                </entity-find>
+                <iterate list="shipmentSourceList" entry="shipmentSourceMap">
+                    <set field="shipmentId" from="shipmentSourceMap.shipmentId"/>
+                </iterate>
+            </iterate>
+            <entity-find entity-name="mantle.shipment.ShipmentItemSource" list="itemSourceList">
+                <econdition field-name="shipmentId"/>
+                <order-by field-name="shipmentPackageSeqId"/>
+            </entity-find>
+            <set field="packageInfoList" from="[]"/>
+            <entity-find entity-name="mantle.shipment.ShipmentPackage" list="packageList">
+                <econdition field-name="shipmentId"/>
+                <order-by field-name="shipmentPackageSeqId"/>
+            </entity-find>
+            <entity-find entity-name="mantle.shipment.ShipmentPackageContent" list="shipmentPackageContentList">
+                <econdition field-name="shipmentId"/>
+                <order-by field-name="productId"/>
+            </entity-find>
+            <iterate list="packageList" entry="pkg">
```

Comment:
Can you explore if we can use any OOTB view entity to get the package contents, like ShipmentPackageContentSummary.